src/SegmentedArray.chpl

```diff
@@ -218,17 +218,16 @@ module SegmentedArray {
         t1 = getCurrentTime();
       }
       var gatheredVals = makeDistArray(retBytes, uint(8));
-      // For comm layer with poor small-message performance, use aggregation
-      // at the expense of memory. Otherwise, unorderedCopy is faster and smaller.
-      if !(CHPL_COMM == 'none' || CHPL_COMM == 'ugni') {
+      // Multi-locale requires some extra localization work that is not needed
+      // in CHPL_COMM=none
+      if CHPL_COMM != 'none' {
```

Comment:
If you want to reduce the number of implementations and test this code for `comm=none` as well, we could just get rid of this branch. 

This will seriously hurt `comm=none` string gather performance, but I'm not sure if that specialization is worthwhile to you or not. On the machine that runs the nightly `chapcs` testing I see performance go from ~2150 MB/s to ~350 MB/s -- https://github.com/mhmerrill/arkouda/issues/527#issuecomment-712400645

Reply:
I still think we need the `comm=none` case.